runtime/compiler/ilgen/Walker.cpp

```diff
@@ -6231,6 +6231,13 @@ TR_J9ByteCodeIlGenerator::genWithField(uint16_t fieldCpIndex)
    TR::Node *passThruNode = TR::Node::create(TR::PassThrough, 1, originalObject);
    genTreeTop(genNullCheck(passThruNode));
 
+   TR_ResolvedJ9Method * owningMethod = static_cast<TR_ResolvedJ9Method*>(_methodSymbol->getResolvedMethod());
+   if (owningMethod->isFieldQType(fieldCpIndex) && owningMethod->isFieldFlattened(comp(), fieldCpIndex, _methodSymbol->isStatic()))
+      {
+      genFlattenableWithFieldWithHelper(fieldCpIndex, newFieldValue, originalObject);
+      return;
+      }
+
```

Comment:
I wonder if `genFlattenableWithFieldWithHelper` should be moved before the stack manipulation at line 6222. `genFlattenableWithFieldWithHelper` will take care of `pop` and `push` and `genNullCheck` instead of piggy back on genWithField here. It'd be easier down the road to debug the code when the stack manipulation inside one function. Actually `loadFlattenableInstanceWithHelper` and `storeFlattenableInstanceWithHelper` already do so.

Reply:
Maybe.

`genWithField()` is a bit different from `loadInstance()` and `storeInstance()` because it only gets called when value types are enabled (i.e. the check is already done by the caller and there isn't anything to do if value types are disabled other than aborting compilation). My thinking was that organizing the code this way reduces code duplication. I'm not sure I see how putting the stack manipulation in one function is going to help with debugging.